packages/typespec-ts/test/integration/multipart.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { assert } from "chai";
+import { resolvePath } from "@typespec/compiler";
+import MultiPartClientFactory, {
+  MultiPartClient
+} from "./generated/payload/multipart/src/index.js";
+import { resolve } from "path";
+// import { readFileSync } from "fs";
+import { readFile } from "fs/promises";
+import { fileURLToPath } from "url";
+// import { createFile } from "@azure/core-rest-pipeline";
+describe.only("MultiPartClient Rest Client", () => {
+  let client: MultiPartClient;
+
+  beforeEach(() => {
+    client = MultiPartClientFactory({
+      allowInsecureConnection: true
+    });
+  });
+
+  it("client should be created", async () => {
+    const root = resolvePath(fileURLToPath(import.meta.url), "../../../temp");
+    console.log(resolve(root, "./assets/image.png"));
+    const path = resolve(root, "./assets/image.png");
+    const file = await readFile(path);
+    // const blob = new File([file]) as any;
+    // blob.name = "image.png";
+    // blob.type = "application/octet-stream";
+    // const images = createFile(file, "image.png", {
+    //   type: "application/octet-stream"
+    // }) as any;
+    const result = await client.path("/multipart/form-data/mixed-parts").post({
+      contentType: "multipart/form-data",
+      body: {
+        id: "123",
+        // profileImage: images
+        profileImage: file
+      }
+    });
```

Comment:
Exception thrown with below:

```
$  rushx integration-test:alone:rlc
Found configuration in C:\Users\marygao\project\autorest.typescript\rush.json

Rush Multi-Project Build Tool 5.97.1 - Node.js 18.12.0 (LTS)
> "cross-env TS_NODE_PROJECT=tsconfig.integration.json mocha -r ts-node/register --experimental-specifier-resolution=node --timeout 4000 ./test/integration/*.spec.ts"

(node:23624) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:23624) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.


  MultiPartClient Rest Client
C:\Users\marygao\project\autorest.typescript\packages\typespec-ts\temp\assets\image.png
    1) client should be created


  0 passing (39ms)
  1 failing

  1) MultiPartClient Rest Client
       client should be created:
     Uncaught Error [ERR_STREAM_PUSH_AFTER_EOF]: stream.push() after EOF
      at new NodeError (node:internal/errors:393:5)
      at readableAddChunk (node:internal/streams/readable:285:30)
      at Readable.push (node:internal/streams/readable:234:10)
      at node:internal/webstreams/adapters:496:22
      at processTicksAndRejections (node:internal/process/task_queues:95:5)



C:\Users\marygao\project\autorest.typescript\common\temp\node_modules\.pnpm\mocha@9.2.2\node_modules\mocha\lib\runner.js:962
    throw err;
    ^

Error: done() called multiple times in test <MultiPartClient Rest Client client should be created> of file C:\Users\marygao\project\autorest.typescript\packages\typespec-ts\test\integration\multipart.spec.ts; in addition, done() received error: AssertionError: expected '400' to equal '204'
    at Context.<anonymous> (file:///C:/Users/marygao/project/autorest.typescript/packages/typespec-ts/test/integration/multipart.spec.ts:39:12)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  showDiff: true,
  actual: '400',
  expected: '204',
  operator: 'strictEqual'
}
    at createMultipleDoneError (C:\Users\marygao\project\autorest.typescript\common\temp\node_modules\.pnpm\mocha@9.2.2\node_modules\mocha\lib\errors.js:428:13)
    at multiple (C:\Users\marygao\project\autorest.typescript\common\temp\node_modules\.pnpm\mocha@9.2.2\node_modules\mocha\lib\runnable.js:290:24)
    at done (C:\Users\marygao\project\autorest.typescript\common\temp\node_modules\.pnpm\mocha@9.2.2\node_modules\mocha\lib\runnable.js:301:14)
    at C:\Users\marygao\project\autorest.typescript\common\temp\node_modules\.pnpm\mocha@9.2.2\node_modules\mocha\lib\runnable.js:377:11
    at processTicksAndRejections (node:internal/process/task_queues:95:5) {
  code: 'ERR_MOCHA_MULTIPLE_DONE',
  valueType: 'object',
  value: AssertionError: expected '400' to equal '204'
      at Context.<anonymous> (file:///C:/Users/marygao/project/autorest.typescript/packages/typespec-ts/test/integration/multipart.spec.ts:39:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5) {
    showDiff: true,
    actual: '400',
    expected: '204',
    operator: 'strictEqual'
  }
}

Node.js v18.12.0
```